Polyconseil/easygettext
# Allow glob paths arguments

Hi, first of all, thank you very much for this great library ! We had lots of troubles with basic xgettext so this really saved our day while using gettext for VueJS.

I would like to propose some changes to the extract CLI. We currently use it in an npm script to extract strings from all files of our app. As we do not particularly want to use the compile CLI (we use a webpack loader to just load the translations we need), we use the extract CLI directly.

The extract CLI needs "exact" file paths. It is unfortunate because I would prefer not to have to change my npm script every time we add a file and so, up until now we just passed a path with a star, for example `src/*.vue`. Our shell expanded it, everyone was happy, the end.

But as our Vue apps get larger, we're starting to have sub-folders, and now we need to get all .vue file paths recursively. This requires options in bash. As a workaround, we could use a package from npm to convert globstar paths for us.

I think nonetheless that it would be sensible for gettext-extract to support paths with globs. What do you think about it ?

I've submitted this PR with an implementation using [fast-glob](https://www.npmjs.com/package/fast-glob). Let me know if something's missing or if the approach I'm taking is wrong or if you have any feedback.

How to test:

 ./src/extract-cli.js <a path with globs> <another path with globs>

It should extract translations from all the files according to the given
glob. If using globstar (**/*) then, it should extract translations from
all files in subdirectories.

Note: it would probably make sense to use streams instead of synchronous files but that's work for another day ;).
